### DISCUSSION TITLE
okhttp 5.0 async calls

### DISCUSSION THREAD

**Discussion by rfc2822** [bitfireAT/dav4jvm#2] [Ideas]
Okhttp 5.0 will support async calls (over Kotlin coroutines) – my idea is that dav4jvm will also support this and then provide async calls, too (which can be used by DAVx5 sync algorithm v2)

**Comment by rfc2822** [comment]
https://square.github.io/okhttp/changelogs/changelog/#version-500-alpha12

> Version 5.0.0-alpha.12
>
> Although this release is labeled alpha, the only unstable thing in it is our new APIs. **This release has many critical bug fixes and is safe to run in production. We’re eager to stabilize our new APIs so we can get out of alpha.**

Maybe not that long anymore ;)

**Comment by ArnyminerZ** [comment]
For ICSx5 right now we have kept using version 4, and added this TODO

https://github.com/ArnyminerZ/icsx5/blob/77ed3253ae9034fb6ac2e5886ca3928710f600d3/app/build.gradle.kts#L133-L134

Since the coroutines dependency only [adds a function](https://github.com/square/okhttp/blob/master/okhttp-coroutines/src/main/kotlin/okhttp3/JvmCallExtensions.kt).

I've copied it into [a separate file](https://github.com/ArnyminerZ/icsx5/blob/dev/app/src/main/java/at/bitfire/icsdroid/AsyncUtils.kt) until the stable 5.0.0 is released. I don't think that function will change much until the stable version is released, and that way we are not using alpha or beta versions.
